src/analysis/calculus/deriv.lean

```diff
@@ -0,0 +1,610 @@
+/-
+Copyright (c) 2019 Gabriel Ebner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner
+-/
+import analysis.calculus.fderiv
+
+/-!
+
+# One-dimensional derivatives
+
+This file defines the derivative of a function `f : 𝕜 → F` where `𝕜` is a
+normed field and `F` is a normed space over this field. The derivative of
+such a function `f` is given by an element `f' : F`.
+
+The theory is developed analogously to the [Fréchet
+derivatives](./fderiv.lean). We first introduce predicates defined in terms
+of the corresponding predicates for Fréchet derivatives:
+
+ - `has_deriv_at_filter f f' x L` states that the function `f` has the
+    derivative `f'` at the point `x` as `x` goes along the filter `L`.
+
+ - `has_deriv_within_at f f' s x` states that the function `f` has the
+    derivative `f'` at the point `x` within the subset `s`.
+
+ - `has_deriv_at f f' x` states that the function `f` has the derivative `f'`
+    at the point `x`.
+
+For the last two notions we also define a functional version:
+
+  - `deriv_within f s x` is a derivative of `f` at `x` within `s`. If the
+    derivative does not exist, then `deriv_within f s x` equals zero.
+
+  - `deriv f x` is a derivative of `f` at `x`. If the derivative does not
+    exist, then `deriv f x` equals zero.
+
+The theorems `fderiv_within_deriv_within` and `fderiv_deriv` show that the
+one-dimensional derivatives coincide with the general Fréchet derivatives.
+
+We also show the existence and compute the derivatives of:
+  - constants
+  - the identity function
+  - linear maps
+  - addition
+  - subtraction
+  - multiplication
+  - negation
+  - multiplication of two functions in `𝕜 → 𝕜`
+  - composition of a function in `𝕜 → F` with a function in `𝕜 → 𝕜`
+
+## Implementation notes
+
+Most of the theorems are direct restatements of the corresponding theorems
+for Fréchet derivatives.
+
+-/
+
+universes u v w
+noncomputable theory
+open_locale classical
+open filter continuous_linear_map asymptotics set
+
+set_option class.instance_max_depth 100
+
+variables {𝕜 : Type u} [nondiscrete_normed_field 𝕜]
+variables {F : Type v} [normed_group F] [normed_space 𝕜 F]
+
+/--
+`f` has the derivative `f'` at the point `x` as `x` goes along the filter `L`.
+
+That is, `f x' = f x + f' • (x' - x) + o(x' - x)` where `x'` converges along the filter `L`.
```

Comment:
Instead of `f' • (x' - x)`, shouldn't this be `(x' - x) • f'` here and below?

Reply:
Ooops.